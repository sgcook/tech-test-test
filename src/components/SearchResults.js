import React from "react";
import PropTypes from "prop-types";
import "../styles/searchResults.css";

function SearchResults({ results }) {
  if(!results.length) {
    return <p>No results</p>
  } else {
   return (
     <div className="search-results__images">
      {results.map((image) => (
        <img className="card-image" src={image} alt="space" />
      ))}
     </div>
   ); 
  }
}

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.array.isRequired,
}