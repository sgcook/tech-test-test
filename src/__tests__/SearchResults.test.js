import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  const validProps = {
    searchResults: ["world", "hello"],
  }

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchResults results={validProps.searchResults}/>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});