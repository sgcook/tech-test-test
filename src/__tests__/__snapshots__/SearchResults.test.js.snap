// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `<DocumentFragment />`;
